### ISSUE TITLE
SDPA fails with sequence lengths not divisible by 32

### ISSUE THREAD

**Issue by jserbedzijaTT** [tenstorrent/tt-metal#32502]
### Describe the bug

The `scaled_dot_product_attention` operation fails when query or key/value sequence lengths are not divisible by 32 and mask is present.

### Steps to reproduce the issue

Run the following test script:

```py
import pytest
import torch
import ttnn

@pytest.mark.parametrize(
    "q_shape,k_shape,v_shape,mask_shape",
    [
        # Query seq_len not divisible by 32
        ((1, 8, 63, 64), (1, 8, 64, 64), (1, 8, 64, 64), (1, 1, 63, 64)),
        # Key/Value seq_len not divisible by 32
        ((1, 8, 64, 64), (1, 8, 50, 64), (1, 8, 50, 64), (1, 1, 64, 50)),
        # Both query and key/value seq_len not divisible by 32
        ((1, 4, 100, 64), (1, 4, 77, 64), (1, 4, 77, 64), (1, 1, 100, 77)),
    ],
)
def test_sdpa_non_divisible_seqlen(q_shape, k_shape, v_shape, mask_shape):
    """Test SDPA with shapes that have sequence lengths not divisible by 32."""
    
    torch.manual_seed(42)
    
    Q = torch.randn(q_shape)
    K = torch.randn(k_shape)
    V = torch.randn(v_shape)
    attention_mask = torch.zeros(mask_shape)
    
    device = ttnn.open_device(device_id=0)
    
    try:
        tt_Q = ttnn.from_torch(Q, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
        tt_K = ttnn.from_torch(K, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
        tt_V = ttnn.from_torch(V, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
        tt_attention_mask = ttnn.from_torch(attention_mask, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
        
        tt_output = ttnn.transformer.scaled_dot_product_attention(
            tt_Q,
            tt_K,
            tt_V,
            is_causal=False,
            attn_mask=tt_attention_mask,
        )
        
        output = ttnn.to_torch(tt_output)
            
    finally:
        ttnn.close_device(device)
```

### Expected behavior

Support padding the sequence dim to be divisible by 32 and then slice the output back to the original dimensions.

**Comment by skhatriTT** [comment]
@cglagovichTT checking in for an update here?

**Comment by cglagovichTT** [comment]
@skhatriTT no update, this is so far unscheduled due priorities of the team

**Comment by skhatriTT** [comment]
Can you please provide a guesstimate on when is this scheduled for? this will help us plan

**Comment by cglagovichTT** [comment]
Sorry I don't have this scheduled. If you would like to take a crack at the issue, I would absolutely welcome a PR

**Comment by skhatriTT** [comment]
Lowering the priority as a workaround is in place

**Comment by jserbedzijaTT** [comment]
Any update on this? @cglagovichTT

**Comment by cglagovichTT** [comment]
@moritztng has a PR which I believe fixes this case https://github.com/tenstorrent/tt-metal/pull/36170
